pkg/network/dhcp/masquerade_test.go

```diff
@@ -77,6 +77,22 @@ var _ = Describe("Masquerade DHCP configurator", func() {
 			return expectedConfig
 		}
 
+		generateExpectedConfigIPv6Disabled := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string) cache.DHCPConfig {
+			ipv4Gateway, ipv4, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv4)
+			expectedConfig := cache.DHCPConfig{Name: ifaceName,
+				IP:                *ipv4,
+				Mtu:               uint16(mtu),
+				AdvertisingIPAddr: ipv4Gateway.IP.To4(),
+				Gateway:           ipv4Gateway.IP.To4(),
+			}
+
+			if macString != nil {
+				mac, _ := net.ParseMAC(*macString)
+				expectedConfig.MAC = mac
+			}
+			return expectedConfig
+		}
```

Comment:
This code is mostly duplicated; would you consider a compose-able approach  ? 

```golang
func generateExpectedConfigOnlyIPv4(...) cache.DHCPConfig {
   	ipv4Gateway, ipv4, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv4)
	expectedConfig := cache.DHCPConfig{Name: ifaceName,
		IP:                *ipv4,
		Mtu:               uint16(mtu),
		AdvertisingIPAddr: ipv4Gateway.IP.To4(),
		Gateway:           ipv4Gateway.IP.To4(),
	}

	if macString != nil {
		mac, _ := net.ParseMAC(*macString)
		expectedConfig.MAC = mac
	}
	return expectedConfig
}

func generateExpectedConfigWithIPv6(...) cache.DHCPConfig {
    ipv6Gateway, ipv6, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv6)
    expectedConfig := generateExpectedConfigOnlyIPv4(...)
    expectedConfig.IPv6 = *ipv6
    expectedConfig.AdvertisingIPv6Addr: ipv6Gateway.IP.To16()
}
```

Reply:
Right. Thanks. I merged the two duplicates into one function with a boolean `ipv6Enabled`

Reply:
Booleans as function attributes are not good, far from it. Better to have 2 functions.

Reply:
OK. Done.
https://github.com/kubevirt/kubevirt/compare/4f0b45fd16ec19310894f412f2e650f9c0c91e4e..909f4bdc5045b3bef8bdc1b1f7df7fc0a0bc7ba3